app/jobs/regular/post_alert.rb

```diff
@@ -3,7 +3,7 @@ class PostAlert < Jobs::Base
 
     def execute(args)
       # maybe it was removed by the time we are making the post
-      if post = Post.find_by(id: args[:post_id])
+      if post = Post.find(args[:post_id])
```

Comment:
This was one of the solution initially but it wasn't solving the root cause of the problem. It'll also introduce a delay as to when the notification is sent out.